dmutils/email.py

```diff
@@ -231,3 +232,95 @@ def decode_invitation_token(encoded_token):
+
+    @staticmethod
+    def get_reference(email_address, template_id, personalisation, time=None, set_time=True, **kwargs):
+        """
+        Method to return the standard reference given the variables the email is sent with.
+
+        :param email_address: Emails recipient
+        :param template_id: Emails template ID on Notify
+        :param personalisation: Template parameters
+        :param time: Generally not wise to set this. Spoofs time that the email was sent.
+        :param set_time: Flag, whether or not to set the time in the reference.
+        :param kwargs: Extra data passed to the reference eg. {'notes': 'Manual resend'}
+        :return: String json 'reference' to be passed to client.send_email_notification or self.send_email
+        """
+        if time is None and set_time:
+            time = datetime.now().strftime(DATETIME_FORMAT)
+        details_dict = OrderedDict([
+            ('email_address', email_address),
+            ('template_id', template_id),
+            ('personalisation', personalisation),
+            ('time', time),
+        ])
+        details_dict.update([(k, v) for k, v in kwargs.items()])
+        return json.dumps(details_dict)
+
+    @staticmethod
+    def decode_reference(ref, datetime_format=None):
+        """Method to decode our own reference, parse string to datetime etc."""
+        details_dict = json.loads(ref)
+        try:
+            details_dict.update(time=datetime.strptime(details_dict['time'], datetime_format or DATETIME_FORMAT))
+        except TypeError:
+            # This throws a TypeError if time is None.
+            # It will also throw a ValueError if the format doesn't match but that's user error so raise.
+            pass
+        return details_dict
+
+    @staticmethod
+    def get_error_message(email_address, error):
+        """Format a logical error message from the error response."""
+        messages = []
+        message_prefix = 'Error sending message to {email_address}: '.format(email_address=email_address)
+        message_string = '{status_code} {error}: {message}'
+
+        for message in error.message:
+            format_kwargs = {
+                'status_code': error.status_code,
+                'error': message['error'],
+                'message': message['message'],
+            }
+            messages.append(message_string.format(**format_kwargs))
+        return message_prefix + ', '.join(messages)
+
+    def send_email(self, email_address, template_id, personalisation=None, reference=None):
+        """
+        Method to send an email using the Notify api.
+
+        :param email_address: String email address for recipient
+        :param template_id: Template accessible on the Notify account whose  api_key you instantiated the class with
+        :param personalisation: The template variables, dict
+        :param reference: If you want to manually specify a reference you may, not recommended.
```

Comment:
If we don't have a use-case and it's not recommended already, do we need this at all?